pkg/arvo/lib/pull-hook.hoon

```diff
@@ -295,72 +312,310 @@
         =^  cards  pull-hook
           (on-agent:og wire sign)
         [cards this]
-      ?.  ?=([%pull %resource *] t.t.wire)
+      ?:  ?=([%version ~] t.t.wire)
+        =^  [cards=(list card) hook=_pull-hook]  state
+          (take-version:hc src.bowl sign)
+        =.  pull-hook  hook
+        [cards this]
+      ?.  ?=([%pull ?(%unver-resource %resource) *] t.t.wire)
         (on-agent:def wire sign)
       =/  rid=resource
         (de-path:resource t.t.t.t.wire)
-      ?+   -.sign  (on-agent:def wire sign)
-          %kick
-        =^  cards  state
-          (handle-kick:hc rid src.bowl)
-        [cards this]
-      ::
-          %watch-ack
-        ?~  p.sign
-          [~ this]
-        =.  tracking
-          (~(del by tracking) rid)
-        =^  cards  pull-hook
-          (on-pull-nack:og rid u.p.sign)
-        :_  this
-        [give-update cards]
-      ::
-          %fact
-        ?.  =(update-mark.config p.cage.sign)
-          =^  cards  pull-hook
-            (on-agent:og wire sign)
-          [cards this]
-        :_  this
-        ~[(update-store:hc rid q.cage.sign)]
-      ==
-      ++  on-leave
-        |=  =path
-        ^-  [(list card:agent:gall) agent:gall]
-        =^  cards  pull-hook
-          (on-leave:og path)
-        [cards this]
-      ::
-      ++  on-arvo
-        |=  [=wire =sign-arvo]
-        ^-  [(list card:agent:gall) agent:gall]
-        =^  cards  pull-hook
-          (on-arvo:og wire sign-arvo)
-        [cards this]
-      ++  on-fail
-        |=  [=term =tang]
-        ^-  [(list card:agent:gall) agent:gall]
-        =^  cards  pull-hook
-          (on-fail:og term tang)
-        [cards this]
-      ++  on-peek   
-        |=  =path
-        ^-  (unit (unit cage))
-        ?:  =(/x/dbug/state path)
-          ``noun+(slop !>(state(inner-state *vase)) on-save:og)
-        ?.  =(/x/tracking path)
-          (on-peek:og path)
-        ``noun+!>(~(key by tracking))
+      =/  versioned=?
+        ?=(%resource i.t.t.t.wire)
+      =^  [cards=(list card) hook=_pull-hook]  state
+        tr-abet:(tr-sign:(tr-abed:track-engine:hc rid) sign versioned)
+      =.  pull-hook  hook
+      [cards this]
+    ::
+    ++  on-leave
+      |=  =path
+      ^-  [(list card:agent:gall) agent:gall]
+      =^  cards  pull-hook
+        (on-leave:og path)
+      [cards this]
+    ::
+    ++  on-arvo
+      |=  [=wire =sign-arvo]
+      ^-  [(list card:agent:gall) agent:gall]
+      =^  cards  pull-hook
+        (on-arvo:og wire sign-arvo)
+      [cards this]
+    ::
+    ++  on-fail
+      |=  [=term =tang]
+      ^-  [(list card:agent:gall) agent:gall]
+      =^  cards  pull-hook
+        (on-fail:og term tang)
+      [cards this]
+    ::
+    ++  on-peek   
+      |=  =path
+      ^-  (unit (unit cage))
+      ?:  =(/x/dbug/state path)
+        ``noun+(slop !>(state(inner-state *vase)) on-save:og)
```

Comment:
Why is inner-state being replaced for this output?

Reply:
Because the prettyprinter will print `inner-state` as a vase otherwise, so we bunt inner-state and slop on the actual inner state.